### PR TITLE
Coverity fixes

### PR DESCRIPTION
**Pull request by pavanbalaji**
Fix some of the issues reported by coverity.

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Add comments such that someone without knowledge of the code could understand
* [x] Have read and agree to the Yaksa CLA terms (https://github.com/pmodels/yaksa/wiki/Yaksa-Contributor-License-Agreement)

### CODE REVIEW

File: src/frontend/types/yaksa_blkindx.c [hunk 1]
```diff
@@ -131,11 +133,11 @@ int yaksa_create_indexed_block(int count, int blocklength, const int *array_of_d
     rc = yaksi_create_hindexed_block(count, blocklength, real_array_of_displs, intype, &outtype);
     YAKSU_ERR_CHECK(rc, fn_fail);
 
-    free(real_array_of_displs);
-
     *newtype = outtype->id;
 
   fn_exit:
+    if (real_array_of_displs)
```
Comment (gcongiu) [inline src/frontend/types/yaksa_blkindx.c:139]: do you need this check?

Comment (pavanbalaji) [reply src/frontend/types/yaksa_blkindx.c:139]: Yeah, in the error case, we might end up here before the array is actually allocated.

Comment (gcongiu) [reply src/frontend/types/yaksa_blkindx.c:139]: `free(NULL)` is safe right?

Comment (pavanbalaji) [reply src/frontend/types/yaksa_blkindx.c:139]: No, but we check for that case when freeing the type.

**Pull request merged**